### PR TITLE
Fix long branch name overflows

### LINKED ISSUE #27971: Long name branch/tag display problem in repo home page
### Description

image

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/go-gitea/gitea/issues/27971
Fixes: https://github.com/go-gitea/gitea/pull/28010

Screenshot 2024-04-09 at 00 19 57

Also fixes a similar issue in issue list where CSS was there but not active because of missing `display: block`.

Screenshot 2024-04-09 at 00 18 25

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by yp05327** [comment]
In this file:
image
should also make some changes.

image

**Comment by silverwind** [comment]
Fixed that one as well: 

image

**Comment by silverwind** [comment]
This branch ref in issue list is now also limited in width:

image

File: web_src/css/repo.css [hunk 1]
```diff
@@ -1164,6 +1164,22 @@ td .commit-summary {
...
+  color: var(--color-text-light-2);
+||||||| 18dd9f9a3f
+.repository.compare.pull .title .issue-title {
+  margin-bottom: 0.5rem;
+}
+
+.repository.compare.pull .title .issue-title .index {
+  color: var(--color-text-light-2);
+}
+
```
Comment (silverwind) [inline web_src/css/repo.css]:
```suggestion
```

**Comment by silverwind** [comment]
Same for projects and milestones:

image

**Comment by silverwind** [comment]
I think I found a somewhat generic solution to dropdown overflow, both on button and in the menu.

image
image

**Comment by yp05327** [comment]
Found a new one, maybe we can do this in another PR 🤔 
image

**Comment by silverwind** [comment]
> Found a new one, maybe we can do this in another PR 🤔

That's kind of a different problem. It needs `...`, but it also needs a `title` there. Likely all such dropdowns need `title`.

File: web_src/css/shared/flex-list.css [hunk 2]
```diff
@@ -80,6 +80,7 @@
 .flex-item .flex-item-body a {
   color: inherit;
   overflow-wrap: anywhere;
+  display: inline-flex;
```
Comment (wxiaoguang) [inline web_src/css/shared/flex-list.css]: What's this used for? I guess we shouldn't covert everything to flex aggressively, especially for a general tag `a`.

Comment (silverwind) [reply web_src/css/shared/flex-list.css]: Without it the padding looks broken here. Checking if there are better solutions.

Screenshot 2024-04-15 at 20 57 06

Comment (silverwind) [reply web_src/css/shared/flex-list.css]: Also, even more broken on org PR:

Screenshot 2024-04-15 at 20 59 48

Comment (silverwind) [reply web_src/css/shared/flex-list.css]: Problem is the inner `.truncated-name` is `display: inline-block` for truncation purpose and that causes the expansion. If I remove it, truncation won't work. If I set `display: block` on it, the item will render on multiple lines for org PR.

Comment (silverwind) [reply web_src/css/shared/flex-list.css]: Found the fix with `vertical-align` in https://github.com/go-gitea/gitea/pull/30345/commits/db2043d1f5566220747748e5426f74541fbfce86:

Screenshot 2024-04-15 at 21 06 13

**Comment by silverwind** [comment]
Another fix pushed for this:

Screenshot 2024-04-15 at 21 12 49

**Comment by silverwind** [comment]
Another issue I found is that `overflow:hidden` on buttons did not work correctly because the CSS had `line-height: 1`. I've reset to `normal` and adjusted the vertical padding of buttons to give the same height as before.

image

**Comment by silverwind** [comment]
So I reverted all these "dropdown button" styles and just use `gt-ellipsis` on that button now. Global button changes will come later (and eliminate `line-height: 1`).

image

**Review by wxiaoguang — APPROVED** [review]
Using "button" inside "dropdown" seems an abuse. It could not be  in this PR's scope and be refactored later.

**Comment by silverwind** [comment]
> Using "button" inside "dropdown" seems an abuse. It could not be in this PR's scope and be refactored later.

Indeed, likely it should be https://fomantic-ui.com/modules/dropdown.html#selection.

**Pull request merged**